flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/DataMapManager.java

```diff
@@ -479,7 +479,19 @@ private TargetNode getTargetNode(Node parentNode, String targetField, SemanticMo
         if (matchingNode == null) {
             return null;
         }
-        return new TargetNode(typeSymbol, fieldSplits[fieldSplits.length - 1], matchingNode);
+        return new TargetNode(typeSymbol, getMappingName(fieldSplits), matchingNode);
+    }
+
+    private String getMappingName(String[] names) {
```

Comment:
I'm not sure this function name describes the functionality. Suggestion: `getLastNonNumericName`.

Reply:
Fixed